packages/admin/admin/src/mui/menu/CollapsibleItem.tsx

```diff
@@ -1,48 +1,51 @@
 import { ChevronDown, ChevronRight, ChevronUp } from "@comet/admin-icons";
-import { Collapse, ComponentsOverrides, Fade, List, Menu, SvgIconProps, Theme, Typography } from "@mui/material";
-import { WithStyles, withStyles } from "@mui/styles";
-import * as React from "react";
-import { useMemo, useRef } from "react";
+import { Collapse, ComponentsOverrides, Fade, List, Menu, Theme, Typography, useThemeProps } from "@mui/material";
+import React from "react";
 import { matchPath, useLocation } from "react-router";
 
-import { MenuCollapsibleItemClassKey, styles } from "./CollapsibleItem.styles";
+import { ThemedComponentBaseProps } from "../../helpers/ThemedComponentBaseProps";
+import { CollapsibleIndicator, ItemTitle, MenuCollapsibleItemClassKey, MenuItem, OwnerState, Root } from "./CollapsibleItem.styles";
 import { MenuContext } from "./Context";
-import { MenuItem, MenuItemProps } from "./Item";
+import { MenuItem as CometMenuItem, MenuItemLevel, MenuItemProps } from "./Item";
 import { MenuItemRouterLinkProps } from "./ItemRouterLink";
 
-export interface MenuLevel {
-    level?: 1 | 2 | 3;
-}
-
 export type MenuChild = React.ReactElement<MenuCollapsibleItemProps | MenuItemRouterLinkProps | MenuItemProps>;
 
-export interface MenuCollapsibleItemProps extends MenuItemProps {
+export interface MenuCollapsibleItemProps
+    extends Omit<MenuItemProps, "slotProps">,
+        ThemedComponentBaseProps<{
+            root: "div";
+            menuItem: typeof CometMenuItem;
+            itemTitle: typeof Typography;
+            collapsibleIndicator: "div";
+        }> {
     children: MenuChild | MenuChild[];
     openByDefault?: boolean;
-    openedIcon?: React.JSXElementConstructor<SvgIconProps>;
-    closedIcon?: React.JSXElementConstructor<SvgIconProps>;
     isMenuOpen?: boolean;
+    iconMapping?: {
+        openDropdown?: React.ReactNode;
+        closeDropdown?: React.ReactNode;
+        firstLevelHoverIndicator?: React.ReactNode;
+        subMenuHoverIndicator?: React.ReactNode;
```

Comment:
Wouldn't it be more consistent to use `menuHoverIndicator` and `submenuHoverIndicator` or `firstLevelHoverIndicator` and `otherLevelsHoverIndicator`? Should those icons even be overridable?

Also please use submenu (one word)